### PR TITLE
Issue 88: Stub for calendar module

### PR DESCRIPTION
**Pull request by lbolla**
I created the stub using `make_stub_files.py` and then edited the types by hand.
I ran into an issue with `mypy`: it complains that True and False are not defined. Therefore, I had to define them in the stub.
Also, `python runtests.py` shows lots of errors: I don't know if this is expected.

### CODE REVIEW

**Comment by matthiaskramm** [comment]
Rather than writing
```
withyear:bool=True
```
please write
```
withyear:bool=...
```

Same  with None.

That way, you also don't run into the missing True/False mypy issue.

**Comment by gvanrossum** [comment]
FWIW the issue of missing True/False is because of a recent change to builtin.pyi; there's a matching change needed to mypy but it has not landed yet (https://github.com/python/mypy/pull/1218).

**Comment by JukkaL** [comment]
Note that if you have a `None` default value in code and use `...` as the default in a stub, you should make the argument type `Optional[...]`, as `None` is a valid argument value if the default is `None`.

**Comment by gvanrossum** [comment]
Also PEP 8 and PEP 484 both say you should put spaces around the equal sign for a default when combined with a type. IOW instead of `encoding: Optional[str]=None` please write `encoding: Optional[str] = None`.

**Comment by lbolla** [comment]
Thanks for your comments. I've amended the pull request with another commit.

File: stdlib/2.7/calendar.pyi [hunk 1]
```diff
@@ -0,0 +1,83 @@
+from typing import Any, Iterable, Optional, Tuple
+import datetime
+
+YearType = int
+MonthType = int
+DayType = int
+WeekdayType = int
```
Comment (gvanrossum) [inline stdlib/2.7/calendar.pyi]: Honestly I don't think these are very useful -- type checkers don't differentiate between them and the way it's written here these names are exported by the stub (but not by the real module). They just lengthen the signatures.

File: stdlib/2.7/calendar.pyi [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+LocaleType = Tuple[Optional[str], Optional[str]]
+
+_colwidth = ...  # type: int
+_spacing = ...  # type: int
```
Comment (gvanrossum) [inline stdlib/2.7/calendar.pyi]: These are internal undocumented constants and shouldn't be defined at all. You can use ... in the signature that mention them.

File: stdlib/2.7/calendar.pyi (hunk 1 above)
Comment (lbolla) [inline stdlib/2.7/calendar.pyi]: But I believe that they provide useful information about what the module's functions do, though. For example, by looking at the signature of a method like `itermonthdays2` one can immediately tell from the return type that the tuples are made of a day and a weekday (and not the other way around) without having to read the docs.

**Comment by lbolla** [comment]
Good point. I've removed mention to the internal variables.

File: stdlib/2.7/calendar.pyi (hunk 1 above)
Comment (gvanrossum) [inline stdlib/2.7/calendar.pyi]: In general I want to resist seeing the type stubs as yet another form of docs. It should serve one need: to provide stubs usable by automatic type checkers. Also the majority of these are redundant, e.g. `month: MonthType` repeated over and over (in fact `MonthType` is *less* helpful than `int`!).

Comment (lbolla) [inline stdlib/2.7/calendar.pyi]: _De gustibus non est disputandum_... I found them useful, but I am happy to remove them to adhere to the project's goals.

**Comment by gvanrossum** [comment]
Thanks! All good now.

**Pull request merged**